crypto/engine/eng_cnf.c

```diff
@@ -130,6 +132,7 @@ static int int_engine_configure(const char *name, const char *value, const CONF
     }
     if (e && (do_init == -1) && !int_engine_init(e)) {
         ecmd = NULL;
+        e = NULL;   /* This is freed by a failed int_engine_init() */
```

Comment:
And this is wrong as well, because the ENGINE_free() must be called regardless of whether the ENGINE_finish() was called in int_engine_init() or not.

Reply:
This too.